blog/2023-01-10-nushell_0_74.md

```diff
@@ -0,0 +1,226 @@
+
+## `$in` now works in `catch` closures
+
+`$in` in `catch` closures now behaves identically to how it does in other closures, like those given to `each` (it's equivalent to what would be the first named argument).
+
+`try { 'x' | math abs } catch { print $in }` behaves the same as `try { 'x' | math abs } catch {|e| print $e }`.
+
+## MIME-types are supported in `ls` with an additional flag. ([fdncred](https://github.com/nushell/nushell/pull/7616))
+
+To find out what application your operating system associates with a particular file, you can now use the `--mime-type` or `-m` flag on our `ls` command.
+This simplifies filtering for particular files and can help you dispatch files to particular programs.
+When opening files with Nushell directly, `open` will still follow the same heuristics using file endings and the built-in `from ...` command parsers.
+
+## Regular expression queries are cached for performance ([rgwood](https://github.com/nushell/nushell/pull/7587))
+
+The primary motivation for this is to make regex and `=~` operator uses in hooks and `color_config` closures more performant.
+
+## All built-in commands now declare their pipeline input and output types ([sholderbach](https://github.com/nushell/nushell/pull/7532))
+
+A few releases back commands internally got the capability to declare not only the types of parameters but also pairs for the input and output on the pipeline.
+With this release we finally declare those input and output types for all core nushell commands.
+This can help you as user to see what a command expects from the pipeline and might return.
+We are exploring how nushell can leverage that for more useful diagnostics and completions. In the future we may introduce syntax for user-defined commands to declare their input and output types explicitly.
+
+![help for wrap command shows I/O type signatures](../assets/images/0_74_io_signatures.png)
+
+# Breaking changes
+
+- `columns` will now error for data that is not a table or a record ([#7593](https://github.com/nushell/nushell/pull/7593))
+- The dataframe specific command `fill-na` has been renamed to `fill-nan` to better represent its capabilities ([#7565](https://github.com/nushell/nushell/pull/7565))
+- The requirements for the names of a nu command and a command alias have been tightened to avoid some problems ([#7392](https://github.com/nushell/nushell/pull/7392))
+- `to toml` will only produce output for records now as the output for tables is ambiguous and may be invalid to parse ([#7597](https://github.com/nushell/nushell/pull/7597))
+- `exec` and known externals don't throw errors when unknowns parameters are passed to them
+- `use`, `overlay use`, `source`, and `source-env` now accept constants defined with `const` as arguments
+- `help` command has been thoroughly refactored and includes several more subcommands 
+- command names in `help commands` and `$nu.scope.commands` are displayed correctly if they were imported from a module with a prefix
```

Comment:
Don't know if I would classify those improvements as breaking. Maybe we need a section for the lightning round of further changes.